proposals/NNNN-ComparisonReform.md

```diff
@@ -0,0 +1,531 @@
+
+* It can be passed to a higher-order function
+* Tuples can implement it
+
+In our opinion, these aren't serious problems, especially in the long term. 
+
+Passing `<=>` as a higher order function basically allows types that aren't
+Comparable, but do provide `<=>`, to be very ergonomically handled by
+algorithms which take an optional ordering function. Types which provide the
+comparable operators but don't conform to Comparable are only pervasive due to
+the absence of conditional conformance. We shouldn't be designing our APIs
+around the assumption that conditional conformance doesn't exist.
+
+When conditional conformance is implemented, the only
+should-be-comparable-but-aren't types that will remain are tuples, which we
+should potentially have the compiler synthesize conformances for.
+
+Similarly, it should one day be possible to extend tuples, although this is a
+more "far future" matter. Until then, the `(T, T) -> Bool` predicate will
+always also be available, and `<` can be used there with the only downside
+being a potential performance hit.
+
+
+### Just Leave Floats Alone
+
+The fact that sorting floats leads to a mess, and storing floats can lead to
+memory leaks and data loss isn't acceptable.
+
+### Just Make Floats Only Have A Total Order
+
+This was deemed too surprising for anyone familiar with floats from any other
+language. It would also probably break a lot more code than this change will.
+
+
+### Just Make Floats Not Comparable
+
+Although floats are more subtle than integers, having places where integers
+work but floats don't is a poor state of affairs. One should be able to sort an
+array of floats and use floats as keys in data structures, even if the latter
+is difficult to do correctly.
+
+### PartialComparable
+
+PartialComparable would essentially just be Comparable without any stated
+ordering requirements, that Comparable extends to provide ordering
+requirements. This would be a protocol that standard IEEE comparison could
+satisfy, but in the absence of total ordering requirements, PartialComparable
+is effectively useless. Either everyone would consume PartialComparable (to
+accept floats) or Comparable (to have reasonable behaviour).
+
+The Rust community adopted this strategy to little benefit. The Rust libs team
+has frequently considered removing the distinction, but hasn't because doing it
+backwards compatibly would be complicated. Also because merging the two would
+just lead to the problems Swift has today.
+
+### Different Names For `compare` and `ComparisonResult`
+
+A few different variants for `ComparisonResult` and its variants were considered:
+
+* Dropping the `ordered` part of `ComparisonResult`'s cases e.g. `.ascending`
+* Naming of `ComparisonResult` as `SortOrder`
+* `enum Ordering { case less, equal, greater }` ([as used by Rust](https://doc.rust-lang.org/std/cmp/enum.Ordering.html))
+* Case values of `inOrder`, `same`, `outOfOrder`
```

Comment:
I agree `inOrder` is a bad name since it doesn't specify *which* order (i.e. ascending, descending)